pilot/cmd/pilot-agent/status/server.go

```diff
@@ -144,33 +124,35 @@ func (s *Server) handleReadyProbe(w http.ResponseWriter, _ *http.Request) {
 	s.mutex.Unlock()
 }
 
-func (s *Server) handleAppReadinessProbe(w http.ResponseWriter, req *http.Request) {
-	requestStatusCode(fmt.Sprintf("http://127.0.0.1%s", s.appReadyURL), w, req)
-}
-
-func (s *Server) handleAppLivenessProbe(w http.ResponseWriter, req *http.Request) {
-	requestStatusCode(fmt.Sprintf("http://127.0.0.1%s", s.appLiveURL), w, req)
-}
-
-func requestStatusCode(appURL string, w http.ResponseWriter, req *http.Request) {
+func (s *Server) handleAppProbe(w http.ResponseWriter, req *http.Request) {
+	appPort := req.Header.Get(IstioAppPortHeader)
+	if _, err := strconv.Atoi(appPort); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
 	httpClient := &http.Client{
 		// TODO: figure out the appropriate timeout?
 		Timeout: 10 * time.Second,
 	}
-
-	appReq, err := http.NewRequest(req.Method, appURL, req.Body)
+	path := req.URL.Path
+	if !strings.HasPrefix(req.URL.Path, "/") {
+		path = "/" + req.URL.Path
+	}
```

Comment:
Can't path.Join do this? Though this is ok.

Reply:
No. It can't because path.Join("http://a", "b") will return "http:/a/b"...

Reply:
ok,